analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -39,8 +39,9 @@ public sealed record ProgramState : IEquatable<ProgramState>
         private ImmutableDictionary<int, int> VisitCount { get; init; }
         private ImmutableDictionary<CaptureId, IOperation> CaptureOperation { get; init; }
         private ImmutableHashSet<ISymbol> PreservedSymbols { get; init; }
+        private ImmutableStack<ExceptionState> Exceptions { get; init; }
 
-        public ExceptionState Exception { get; private init; }
+        public ExceptionState Exception => Exceptions.IsEmpty ? null : Exceptions.Peek();
```

Comment:
So we keep checking only the last exception and we keep the rest for when we return to the parent  try/catch block.

Reply:
Exactly